File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_idxnlprn.py
"""Reference emit for IDXNLPRN — INDEXED START NOT LESS + READ PREV → 45.50."""
# EXPECTED: 45.50
ROWS = {10: 12.50, 20: 8.00, 42: 25.00, 55: 30.00, 99: 15.00}
START = 42
# GnuCOBOL: after START NOT LESS, READ PREVIOUS includes the positioned key.
total = sum(v for k, v in ROWS.items() if int(k) <= START)
print(f"{total:.2f}")

File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_idxnlprn.cs
// EXPECTED: 45.50
// Reference emit for IDXNLPRN — INDEXED START NOT LESS + READ PREV.
using System;
using System.Collections.Generic;
public static class IdxnlprnRef {
  public static void Main() {
    var rows = new Dictionary<int, double> {
      [10] = 12.50, [20] = 8.00, [42] = 25.00, [55] = 30.00, [99] = 15.00
    };
    int start = 42;
    double total = 0;
    foreach (var e in rows) {
      if (e.Key <= start) total += e.Value;
    }
    Console.WriteLine(total.ToString("0.00"));
  }
}
